server/s3_motr_reader.h

```diff
@@ -150,6 +152,15 @@ class S3MotrReader {
 
   S3MotrReaderOpState state;
 
+  // md5 for the content read from motr.
+  // In case of it's a GET for multipart upload it's calculated for each part
```

Comment:
Do we also do a checksum of checksums for the entire object or that's not necessary?

Reply:
Multipart upload is done in the following way: parts are uploaded one by one, then there is CompleteMultipartUpload call that is only about changing the metadata. To calculate the entire object md5 checksum we would have to read the entire object during CompleteMultipartUpload. There may be a hash that allows to calculate full hash using partial hashes, but md5 checksums were already implemented in the code for PUT and UploadPart, so sticking with md5 checksums was the easiest way to go.

Reply:
Added this explanation to the code.